app/controllers/projects_controller.rb

```diff
@@ -54,9 +54,19 @@ def create
   end
 
   def show
+    includes = params[:include].present? && params[:include].to_s.split(',') || []
+
+    envgroups = :environment_variable_groups
+    with_scope = :environment_variables_with_scope
+
+    includes.push(envgroups) if includes.include?(envgroups.to_s)
+    includes.push(with_scope) if includes.include?(with_scope.to_s)
+
     respond_to do |format|
       format.html { @stages = @project.stages }
-      format.json { render json: @project.as_json }
+      format.json do
+        render json: @project.as_json(include: includes)
```

Comment:
should not use as_json directly but the render_as_json helper ... which also whitelists allowed includes